lib/test_prof/memory_prof/printer.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+require "test_prof/memory_prof/printer/number_to_human"
+require "test_prof/ext/string_truncate"
+
+module TestProf
+  module MemoryProf
+    class Printer
+      include Logging
+      using StringTruncate
+
+      def initialize(tracker)
+        @tracker = tracker
+      end
+
+      def print
+        messages = [
+          "MemoryProf results\n\n",
+          print_total,
+          print_block("groups", tracker.groups),
+          print_block("examples", tracker.examples)
+        ]
+
+        log :info, messages.join
+      end
+
+      private
+
+      attr_reader :tracker
+
+      def print_total
+        if tracker.alloc?
+          "Total allocations: #{tracker.total_memory}\n\n"
+        else
+          "Final RSS: #{number_to_human(tracker.total_memory)}\n\n"
+        end
+      end
+
+      def print_block(name, items)
+        return if items.empty?
+
+        <<~GROUP
+          Top #{tracker.top_count} #{name} (by #{tracker.mode}):
+
+          #{print_items(items)}
+        GROUP
+      end
+
+      def print_items(items)
+        messages =
+          items.map do |item|
+            <<~ITEM
+              #{item[:name].truncate(30)} (#{item[:location]}) – +#{memory_amount(item)} (#{memory_percentage(item)}%)
+            ITEM
+          end
+
+        messages.join
+      end
+
+      def memory_amount(item)
+        if tracker.rss?
```

Comment:
I think, we should move this logic into a tracker class itself and avoid using predicate methods `#rss?` and `#alloc?`—that makes trackers coupled; not the best design. Imagine adding one more tracker type.

Something like `tracker.value_to_human(item[:memory])` would work here.